src/lib.rs

```diff
@@ -102,12 +103,51 @@ pub fn select_coin_lowestlarger(
 /// Return None, if no solution exists.
 pub fn select_coin_fifo(
     inputs: Vec<OutputGroup>,
-    opitons: CoinSelectionOpt,
+    options: CoinSelectionOpt,
     excess_strategy: ExcessStrategy,
 ) -> Result<Option<(Vec<u32>, WasteMetric)>, SelectionError> {
-    unimplemented!()
+    let mut totalvalue: u64 = 0;
+    let mut totalweight: u32 = 0;
+    let mut selectedinputs: Vec<u32> = Vec::new();
+    impl Ord for OutputGroup {
```

Comment:
[sort_by_key](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort_by_key) method could be used here.
Implementing a trait for a type is for re-usability across the code base.

Reply:
1. Implemented the sorting using sort_by_key
2. Moved the impl Ord out of the FIFO function to below the struct definition